pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+        if weight:
+            weight_bytes_value = bytes(weight, "ascii")
+        else:
+            weight_bytes_value = None
+
+        if enabled:
+            enabled_bytes_value = bytes(enabled, "ascii")
+        else:
+            enabled_bytes_value = None
+        status = self._dll.insertOptimizationGoalDefinitionRow(
+            row_index,
+            lower_frequency_bytes_value,
+            upper_frequency_bytes_value,
+            goal_value_bytes_value,
+            condition_bytes_value,
+            parameter_name_bytes_value,
+            weight_bytes_value,
+            enabled_bytes_value,
+        )
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def remove_row(self, row_index):
+        """Remove a row from the optimization goals table.
+
+        Parameters
+        ----------
+        row_index: int
+            Index of the row. Valid values range from ``0`` to ``49``, inclusive.
+        """
+        status = self._dll.removeOptimizationGoalDefinitionRow(row_index)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def set_design_goals(self):
+        """Set the updated design goals definition."""
+        status = self._dll.designGoals()
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def save_goals(self, design, file_path) -> str:
+        """Save the optimization goals from a design's optimization goals table to a CSV file.
+
+        Parameters:
+        ----------
+        design: The design object containing the optimization goals table.
+        file_path: The path to the CSV file where the goals will be saved.
+        """
+        with open(file_path, mode="w", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow(
+                ["Start Frequency", "Stop Frequency", "Goal Value", "Condition", "Parameter", "Weight", "Enabled"]
+            )
+            for row_index in range(design.optimization_goals_table.row_count):
+                row_data = design.optimization_goals_table.row(row_index)
+                writer.writerow(row_data)
+
+    def load_goals(self, file_path) -> str:
+        """Load optimization goals from a CSV file into this optimization goals table.
+
+        Parameters:
+        ----------
+        file_path: The path to the CSV file from which the goals will be loaded.
+        """
+        try:
+            with open(file_path, mode="r", newline="") as file:
+                reader = csv.reader(file)
+                next(reader, None)
+                for row in reader:
+                    self.append_row(*row)
+        except FileNotFoundError:
+            print(f"File {file_path} not found.")
+        except Exception as e:
+            print(f"An error occurred while loading goals: {e}")
+
+    def adjust_goal_frequency(self, adjust_goal_frequency_string):
+        """Adjust the goal frequencies by an entered adjusting frequency value."""
```

Comment:
Can this be a negative frequency to adjust the frequencies downward?
```suggestion
        """Adjust all goal frequencies in the table by the adjusting frequency value which can be positive or negative."""
```